### ISSUE TITLE
About naming your export member $enum?

### ISSUE THREAD

**Issue by gremo** [UselessPickles/ts-enum-util#6]
Just curious, why naming your export member `$enum` in your examples? Does this mean anything?

Anyways, thanks, useful library 👍

**Comment by UselessPickles** [comment]
Thats the name of the exported function. I'm not really naming it `$enum` for the purpose of my examples, but simply importing it exactly as it is exported.

I wanted something short, simple, and easy to remember that was unlikely to conflict with anything else so that no one would have to alias it when importing it. By exporting a clear, memorable, and uniquely named "thing", this allows you to simply start writing code that uses `$enum()` and most IDEs can take care of inserting the `import { $enum } from "ts-enum-util";` for you (either automatically, or with a quick keyboard shortcut).

I ended up using inspiration from the naming of jquery's `$()` function. Many javascript developers are familiar with jquery, and the fact that `$()` gives you a wrapper around a raw DOM element to expose additional/simplified functionality around the DOM element.

Similarly, `$enum()` gives you a wrapper around a raw `enum` to expose additional/simplified functionality around the `enum`.

This isn't the first time I've been asked why I named it `$enum`, so maybe I'll start a FAQ section in the README with this as its first entry.
